servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ClientEffectiveStrategyTest.java

```diff
@@ -44,277 +43,284 @@
 
-    @ParameterizedTest
-    @EnumSource(ClientStrategyCase.class)
-    void streaming(ClientStrategyCase strategyCase) throws Exception {
-        try (Params params = strategyCase.apply(ClientType.AsyncStreaming)) {
-            assertThat("Null params supplied", params, notNullValue());
-            Collection<Buffer> buffers = params.client().request(params.client().get("/"))
-                    .flatMapPublisher(StreamingHttpResponse::payloadBody).toFuture().get();
-            String response = buffersToResponse(buffers, StandardCharsets.US_ASCII);
-            assertThat(response, is(GREETING));
-            params.verifyOffloads();
+    @SuppressWarnings("unused")
+    static Stream<Arguments> casesSupplier() {
+        List<Arguments> arguments = new ArrayList<>();
+        for (ClientType clientType : ClientType.values()) {
+            for (HttpExecutionStrategy builderStrategy : BUILDER_STRATEGIES) {
+                for (HttpExecutionStrategy filterStrategy : FILTER_STRATEGIES) {
+                    for (HttpExecutionStrategy lbStrategy : LB_STRATEGIES) {
+                        for (HttpExecutionStrategy cfStrategy : CF_STRATEGIES) {
+                            arguments.add(Arguments.of(clientType,
+                                    builderStrategy, filterStrategy, lbStrategy, cfStrategy));
+                        }
+                    }
+                }
+            }
         }
+        return arguments.stream();
     }
 
-    @ParameterizedTest
-    @EnumSource(ClientStrategyCase.class)
-    void async(ClientStrategyCase strategyCase) throws Exception {
-        try (Params params = strategyCase.apply(ClientType.Async)) {
-            assertThat("Null params supplied", params, notNullValue());
-            HttpClient httpClient = params.client().asClient();
-            HttpResponse response = httpClient.request(httpClient.get("/")).toFuture().get();
-            assertThat(response.payloadBody().toString(StandardCharsets.US_ASCII), is(GREETING));
-            params.verifyOffloads();
+    @ParameterizedTest(name = "API={0} builder={1} filter={2} LB={3} CF={4}")
+    @MethodSource("casesSupplier")
+    void clientStrategy(ClientType clientType,
+                  @Nullable final HttpExecutionStrategy builderStrategy,
+                  @Nullable final HttpExecutionStrategy filterStrategy,
+                  @Nullable final HttpExecutionStrategy lbStrategy,
+                  @Nullable final HttpExecutionStrategy cfStrategy) throws Exception {
+        HttpExecutionStrategy effectiveStrategy = null == builderStrategy ?
+                defaultStrategy() : builderStrategy;
+        effectiveStrategy = null == filterStrategy || anyStrategy() == filterStrategy ||
+                defaultStrategy() == filterStrategy || noOffloadsStrategy() == filterStrategy ?
+                effectiveStrategy : effectiveStrategy.merge(filterStrategy);
+        effectiveStrategy = null == lbStrategy || anyStrategy() == lbStrategy ||
```

Comment:
can you put this logic in a function to share code and provide the operation a name for ease of understanding?